scan_counter.py

```diff
@@ -55,10 +73,20 @@ def __init__(self):
 
             def run(self):
                 while not self.stop:
-                    sinargs = np.arange(self.idx, self.idx+SAMPLES_PER_BIN) * 20*np.pi / BUFFER_SIZE
-                    sin = np.sin(sinargs) * 2000
-                    sin += (np.random.rand(SAMPLES_PER_BIN) - 0.5) * 20
-                    buf[self.idx:self.idx+SAMPLES_PER_BIN*2:2] = (sin + 250) / SAMPLES_PER_BIN
+                    buf[self.idx:self.idx+SAMPLES_PER_BIN*2:2] = 0
+                    bins = np.zeros(SAMPLES_PER_BIN, dtype=np.int64)
+                    num_burst = 1 if np.random.poisson() else 0
+                    l = np.random.randint(0,SAMPLES_PER_BIN, size=num_burst)
+                    s = np.random.poisson(30, size=num_burst)
+                    ph_locs = np.random.normal(scale=50, size=s).astype(int) + l
+                    for loc in ph_locs:
+                        if loc < 0 or loc >= bins.size:
+                            continue
+                        buf[self.idx+2*loc] += 1
+                    # sinargs = np.arange(self.idx, self.idx+SAMPLES_PER_BIN) * 20*np.pi / BUFFER_SIZE
```

Comment:
I assume the new mock-data is better suited for autocorrelation?

Please delete the old code (that you already commented out)

Reply:
Yes, and in my latest additions, I've improved it even more...